networkit/cpp/graph/GraphTools.cpp

```diff
@@ -201,14 +201,17 @@ double density(const Graph &G) noexcept {
 }
 
 double volume(const Graph &G) {
-    if (G.isDirected())
-        throw std::runtime_error("Volume computation is only supported for undirected graphs.");
+    double volume = 0.0;
 
-    if (!G.isWeighted()) {
-        return 2.0 * G.numberOfEdges();
+    if (G.isWeighted()) {
+        volume = G.totalEdgeWeight();
     } else {
-        return 2.0 * G.totalEdgeWeight();
+        volume = G.numberOfEdges();
+    }
```

Comment:
`G.totalEdgeWeight()` already handles the weighted/unweighted case, you can simply use `double volume = G.totalEdgeWeight()` and drop the `if`.